Core/Core/ActAsUser/ActAsUserViewController.swift

```diff
@@ -0,0 +1,129 @@
+
+        domainTextField.addTarget(self, action: #selector(enterPressed), for: .editingDidEndOnExit)
+        userIDTextField.addTarget(self, action: #selector(enterPressed), for: .editingDidEndOnExit)
+
+        redPanda.transform = redPanda.transform.rotated(by: .pi / -6).translatedBy(x: 0, y: 0)
+        actAsUserButton.roundCorners(corners: [.topLeft, .topRight, .bottomLeft, .bottomRight], radius: 5)
+        animatePanda()
+    }
+
+    func animatePanda() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [weak self] in
+            self?.redPandaTopConstraint.constant -= 30
+            self?.redPandaTrailingConstraint.constant -= 82
+            UIView.animate(withDuration: 1.5, animations: { [weak self] in
+                guard let redPanda = self?.redPanda else {
+                    return
+                }
+
+                redPanda.transform = redPanda.transform
+                    .rotated(by: .pi / 6)
+                self?.view.layoutIfNeeded()
+                }, completion: { [weak self] _ in
+                    self?.redPandaTopConstraint.constant += 30
+                    self?.redPandaTrailingConstraint.constant += 82
+                    UIView.animate(withDuration: 1.5, delay: 5, animations: { [weak self] in
+                        guard let redPanda = self?.redPanda else {
+                            return
+                        }
+                        redPanda.transform = redPanda.transform
+                            .rotated(by: .pi / -6)
+                        self?.view.layoutIfNeeded()
+                    }, completion: { [weak self] _ in
+                        self?.animatePanda()
+                    })
+            })
+        }
+    }
+
+    func showMasqueradingError() {
+        let alert = UIAlertController(
+            title: NSLocalizedString("Error", bundle: .core, comment: ""),
+            message: NSLocalizedString("There was an error with Act as User. Please try again later.", bundle: .core, comment: ""),
+            preferredStyle: .alert
+        )
+        alert.addAction(UIAlertAction(title: NSLocalizedString("Cancel", bundle: .core, comment: ""), style: .cancel, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+
+    @IBAction func actAsUserPressed(_ sender: UIButton) {
+        guard let domain = self.domainTextField.text, let userID = self.userIDTextField.text else {
+            return
+        }
+        presenter?.didSubmit(domain: domain, userID: userID) { [weak self] err in
+            if err == nil {
+                self?.dismiss(animated: true, completion: nil)
+            } else {
+                self?.showMasqueradingError()
+            }
+        }
+    }
+}
+
+extension ActAsUserViewController {
+    @objc
+    func updateActAsUserButtonDisabledStatus() {
+        self.actAsUserButton.isEnabled = domainTextField.text?.count ?? 0 > 0 && userIDTextField.text?.count ?? 0 > 0
```

Comment:
`.text?.isEmpty == false` No need to count the characters, since we only care if it's empty or not.

Reply:
Oh great suggestion.